doc/monads/intro.md

```diff
@@ -42,6 +42,12 @@ This section introduces the `StateM` monad. This monad allows you to access a pa
 both read from and write to. It opens the door to fully stateful programming, allowing you to do many
 of the things a function programming language supposedly "can't" do.
 
+## [ST Monad](st.lean.md)
+This section introduces the `ST` monad. This monad allows you to use updatable state in a
+ReaderM context. This is an alternative to `StateM` that is more convenient in some scenarios,
+specifically in multi-threading scenarios because this monad ensures single threaded access to
+the mutable state.
```

Comment:
This is the chapter on Monads where the reader just learned about ReaderM and StateM and is now seeing that an updatable ST.Ref can be used in a ReaderM context which provides a kind of hybrid model somewhere in between ReaderM and StateM.  This is why I have this kind of language and at least this made sense to me when I try to fit all this info in my brain.

I think your comments here may indicate we need a separate doc on "References" that drops all the connection to Monads?

Reply:
Not necessarily. References and the ST monad are pretty tightly related, and IO references make sense to document in the same section since they are basically the same thing, and it would be way too much to have a section on the IO monad similar to the others. You can't talk about references without talking about monads because the functions are all monadic and you need the do notation to present the examples. Talking about ST *is* talking about references. The stuff with asynchronous programming on the other hand is a distraction, ST doesn't have anything to do with tasks or threads and indeed it doesn't even mix very well.